File: src/leetcode/2025/12/Solution412.java
import java.util.List;

public class Solution412 {
    public List<String> fizzBuzz(int n) {
        List<String> result = new java.util.ArrayList<>();
        for (int i = 1; i <= n; i++) {
            if (i % 3 == 0 && i % 5 == 0) {
                result.add("FizzBuzz");
            } else if (i % 3 == 0) {
                result.add("Fizz");
            } else if (i % 5 == 0) {
                result.add("Buzz");
            } else {
                result.add(Integer.toString(i));
            }
        }
        return result;
    }

    public static void main(String[] args) {
        Solution412 solution = new Solution412();

        int n1 = 15;
        List<String> result1 = solution.fizzBuzz(n1);
        System.out.println("FizzBuzz up to " + n1 + ": " + result1);

        int n2 = 5;
        List<String> result2 = solution.fizzBuzz(n2);
        System.out.println("FizzBuzz up to " + n2 + ": " + result2);

        int n3 = 1;
        List<String> result3 = solution.fizzBuzz(n3);
        System.out.println("FizzBuzz up to " + n3 + ": " + result3);

        int n4 = 3;
        List<String> result4 = solution.fizzBuzz(n4);
        System.out.println("FizzBuzz up to " + n4 + ": " + result4);
    }
}

File: src/leetcode/2025/12/Solution412.cpp
#include <vector>
#include <string>
#include <iostream>
using namespace std;

class Solution412 {
public:
    vector<string> fizzBuzz(int n) {
        vector<string> result;
        for (int i = 1; i <= n; ++i) {
            if (i % 3 == 0 && i % 5 == 0) {
                result.push_back("FizzBuzz");
            } else if (i % 3 == 0) {
                result.push_back("Fizz");
            } else if (i % 5 == 0) {
                result.push_back("Buzz");
            } else {
                result.push_back(to_string(i));
            }
        }
        return result;
    }
};

int main() {
    Solution412 sol;

    int n1 = 3;
    vector<string> fizzBuzzResult1 = sol.fizzBuzz(n1);
    for (const string& s : fizzBuzzResult1) {
        cout << s << " ";
    }
    cout << endl;

    int n2 = 5;
    vector<string> fizzBuzzResult2 = sol.fizzBuzz(n2);
    for (const string& s : fizzBuzzResult2) {
        cout << s << " ";
    }
    cout << endl;

    int n3 = 15;
    vector<string> fizzBuzzResult3 = sol.fizzBuzz(n3);
    for (const string& s : fizzBuzzResult3) {
        cout << s << " ";
    }
    cout << endl;

    return 0;
}